test/DynamicFees.t.sol

```diff
@@ -149,6 +150,130 @@ contract TestDynamicFees is Test, Deployers, GasSnapshot {
         assertEq(_fetchPoolSwapFee(key), 123);
     }
 
+    function test_updateDynamicSwapFee_100PercentFee_AmountIn() public {
+        assertEq(_fetchPoolSwapFee(key), 0);
+
+        dynamicFeesHooks.setFee(1000000);
+
+        IPoolManager.SwapParams memory params =
+            IPoolManager.SwapParams({zeroForOne: true, amountSpecified: -100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
```

Comment:
would love for these tests to have a higher amountSpecified. Just because this is swapping 100 wei of a token which is ~nothing haha. Maybe 1e18?

Reply:
And same with all of them. Or even 123456789 or something so we can see what happens with odd numbers where more rounding will happen - e.g. the test below takes 50%

Reply:
still would love this 👀